src/validator.cpp

```diff
@@ -1635,6 +1708,530 @@ void Validator::ValidatorImpl::validateMathMLElements(const XmlNodePtr &node, co
+
+        auto parentNode = node->parent();
+        auto siblingCount = hasOneOrTwoMathmlSiblings(parentNode, node, component);
+
+        if ((siblingCount != 0)
+            && isFirstMathmlChild(parentNode, node, component)) {
+            (siblingCount == 2)
+                && hasFirstMathmlSiblingWithName(parentNode, node, "logbase", component);
+        }
+    } else if (node->isMathmlElement("ceiling")
+               || node->isMathmlElement("floor")) {
+        auto parentNode = node->parent();
+
+        hasOneMathmlSibling(parentNode, node, component)
+            && isFirstMathmlChild(parentNode, node, component);
+    } else if (node->isMathmlElement("min")) {
+    } else if (node->isMathmlElement("max")) {
+    } else if (node->isMathmlElement("rem")) {
+        // Calculus elements.
+
+    } else if (node->isMathmlElement("diff")) {
+        auto parentNode = node->parent();
+
+        hasTwoMathmlSiblings(parentNode, node, component)
+            && isFirstMathmlChild(parentNode, node, component)
+            && hasFirstMathmlSiblingWithName(parentNode, node, "bvar", component);
+
+        // Trigonometric operators.
+
+    } else if (node->isMathmlElement("sin")
+               || node->isMathmlElement("cos")
+               || node->isMathmlElement("tan")
+               || node->isMathmlElement("sec")
+               || node->isMathmlElement("csc")
+               || node->isMathmlElement("cot")
+               || node->isMathmlElement("sinh")
+               || node->isMathmlElement("cosh")
+               || node->isMathmlElement("tanh")
+               || node->isMathmlElement("sech")
+               || node->isMathmlElement("csch")
+               || node->isMathmlElement("coth")
+               || node->isMathmlElement("arcsin")
+               || node->isMathmlElement("arccos")
+               || node->isMathmlElement("arctan")
+               || node->isMathmlElement("arcsec")
+               || node->isMathmlElement("arccsc")
+               || node->isMathmlElement("arccot")
+               || node->isMathmlElement("arcsinh")
+               || node->isMathmlElement("arccosh")
+               || node->isMathmlElement("arctanh")
+               || node->isMathmlElement("arcsech")
+               || node->isMathmlElement("arccsch")
+               || node->isMathmlElement("arccoth")) {
+        auto parentNode = node->parent();
+
+        hasOneMathmlSibling(parentNode, node, component)
+            && isFirstMathmlChild(parentNode, node, component);
+
+        // Piecewise statement.
+
+    } else if (node->isMathmlElement("piecewise")) {
+        for (size_t i = 0, iMax = mathmlChildCount(node); i < iMax; ++i) {
+            validateMathMLElementsChildrenAndSiblings(mathmlChildNode(node, i), component);
+        }
+    } else if (node->isMathmlElement("piece")) {
+        if (hasTwoMathmlChildren(node, component)) {
+            for (size_t i = 0, iMax = mathmlChildCount(node); i < iMax; ++i) {
```

Comment:
is a loop necessary given we know there are exactly two MathML children to validate?

Reply:
Overuse of copying/pasting. My bad!

1bf3812f